packages/hardhat-ts/contracts/DAMM/Redemption.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "../openzeppelin/Ownable.sol";
+import "../openzeppelin/ERC20.sol";
+import "../openzeppelin/ReentrancyGuard.sol";
+
+// from and to are addresses of tokens who have the same decimals
+contract Redemption is Ownable, ReentrancyGuard {
+    ERC20 constant BDAMM = ERC20(0xfa372fF1547fa1a283B5112a4685F1358CE5574d);
+    ERC20 constant DAMM = ERC20(0xb3207935FF56120f3499e8aD08461Dd403bF16b8);
+    ERC20 constant USDC = ERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+
+    uint256 redemptionUSDCPrice_;
+
+    constructor(uint256 _redemptionUSDCPrice) {
+      redemptionUSDCPrice_ = _redemptionUSDCPrice;
+    }
+
+    /* 
+      amount is the amount of BDAMM to swap to DAMM 1:1 (18 decimals)
+      redemptionFee is the amount of USDC to send with the redemption (6 decimals)
+    */
+    function swap(uint256 amount, uint256 redemptionFee) nonReentrant external {
+        address user = msg.sender;
```

Comment:
Possible improvement of msg.sender to _msgSender() for https://docs.openzeppelin.com/contracts/4.x/api/metatx#ERC2771Context-_msgSender--